### PR TITLE
amp-script: fix long task start/end order for within the same macrotask

### PR DESCRIPTION
**Pull request by samouri**
**summary**
Fixes https://github.com/ampproject/amp.dev/issues/4879, https://github.com/ampproject/amphtml/issues/30548

The activity tracker relies on careful ordering of `longTask` events. We need it to be a sequence of START --> END --> START --> END.

The `.catch` call ended up delaying the `LONG_TASK_END` event by an extra microtask, which was just enough to make it occur after a 2nd start event.

**before**
Screen Shot 2021-04-06 at 12 16 46 PM

**after**
Screen Shot 2021-04-06 at 12 17 37 PM

**note**: a more robust but larger fix would either be: (a) start/end counter (so order doesn't matter), or (b) the `longTask` fn gets called with a "type" for start/end (avoids promises)

### CODE REVIEW

**Review by caroqliu — APPROVED** [review]

File: extensions/amp-script/0.1/user-activation-tracker.js [hunk 1]
```diff
@@ -100,13 +100,13 @@ export class UserActivationTracker {
       return;
     }
     this.inLongTask_ = true;
```
Comment (caroqliu) [inline extensions/amp-script/0.1/user-activation-tracker.js:102]: Is it worth checking if this is already `true`? Should we be able to assert on it?

Comment (samouri) [reply extensions/amp-script/0.1/user-activation-tracker.js:102]: What does that buy us?

Comment (caroqliu) [reply extensions/amp-script/0.1/user-activation-tracker.js:102]: Based on this from the PR description:
We need it to be a sequence of START --> END --> START --> END.

Would it be possible or problematic if we had START --> START --> END --> END?

Comment (samouri) [reply extensions/amp-script/0.1/user-activation-tracker.js:102]: Yes! That is the exact problem scenario. We could do a devAssert as a form of documentation for ourselves?

Comment (caroqliu) [reply extensions/amp-script/0.1/user-activation-tracker.js:102]: SG, let's do that!

Comment (samouri) [reply extensions/amp-script/0.1/user-activation-tracker.js:108]: Updated with the `devAssert`

**Pull request merged**